server/handlers/project_command_output_handler.go

```diff
@@ -83,32 +93,47 @@ func NewAsyncProjectCommandOutputHandler(
 		projectStatusUpdater:   projectStatusUpdater,
 		projectJobURLGenerator: projectJobURLGenerator,
 		projectOutputBuffers:   map[string][]string{},
+		pullToJobMapping:       map[models.PullContext]map[string]bool{},
 	}
 }
 
 func (p *AsyncProjectCommandOutputHandler) Send(ctx models.ProjectCommandContext, msg string) {
 	p.projectCmdOutput <- &models.ProjectCmdOutputLine{
-		ProjectInfo: ctx.PullInfo(),
-		Line:        msg,
+		JobID: ctx.JobID,
+		JobContext: models.JobContext{
+			HeadCommit: ctx.Pull.HeadCommit,
+			PullContext: models.PullContext{
+				PullNum:     ctx.Pull.Num,
+				Repo:        ctx.BaseRepo.Name,
+				ProjectName: ctx.ProjectName,
+				Workspace:   ctx.Workspace,
+			},
+		},
+		Line: msg,
 	}
 }
 
-func (p *AsyncProjectCommandOutputHandler) Register(projectInfo string, receiver chan string) {
-	p.addChan(receiver, projectInfo)
+func (p *AsyncProjectCommandOutputHandler) Register(jobID string, receiver chan string) {
+	p.addChan(receiver, jobID)
 }
 
 func (p *AsyncProjectCommandOutputHandler) Handle() {
 	for msg := range p.projectCmdOutput {
 		if msg.ClearBuffBefore {
-			p.clearLogLines(msg.ProjectInfo)
+			p.clearLogLines(msg.JobID)
```

Comment:
we don't need this anymore, we are generating new JobID for each new command, right?

Reply:
Hmm true! Good catch :)

Reply:
Removed!